### DISCUSSION TITLE
gonb is not working as i wished

### DISCUSSION THREAD

**Discussion by encryptblockr** [janpfeifer/gonb#67] [General]
with python in jupyter, whatever python script you have runs as is even when you take that script and convert to .py file and run it

why cant we do the same with golang with jupyter
seems gonb needs special attention and thus the same code can not run as it does with python

that defeats the whole purpose as we need to re-write code whenever we getting out of gonb

**Comment by janpfeifer** [comment]
hey @encryptblockr ,

Sorry I don't see your point ... Usually I don't convert the jupyter notebooks (Python or Go) to code directly. The use cases I see folks doing are:

1. They do development/exeperiments in Jupyter notebook, and later they share the results in a notebook itself.
2. Development/experiments in a notebook, and when happy with the result (let's say a ML model code), later one copy & paste the relevant parts to library.

Now, outside of the browser, there is also `nbconvert` tool, which executes a notebook (again, it works for both Python and Go) and outputs the results as a HTML/Ascii/PDF/... page (you can configure the output). Would that be what you are looking for ?

cheers

**Reply by encryptblockr** [reply]
what am looking for is a way to test out multiple scripts and i can just copy and use the scirpts or codes as-is without having to modify the code to run outside of gonb

with python, i dont have to re-write the code is i want to run like python name-of-script.py but with this gonb i will have to re-write the whole code to the go way and away from the gonb way

**Reply by janpfeifer** [reply]
Would you mind showing us an example in Python ? I'm not understanding ...

Just in case that's what you mean: If you have the Go code somewhere else (let's say a local directory), you can import it in GoNB, and use it as an external package so to say. 

This will not work in Go (unrelated to notebooks) if the package you want to execute is the `main` package. But it's easy enough to execute binaries (Go or otherwise) from Go, using the [os/exec](https://pkg.go.dev/os/exec) library. There is also the shortcut `!<shell_command>` to execute the command using shell.

Example of using the shell shortcut:

```
!cd /path/to/my/go/script && go run .
```

**Comment by oderwat** [comment]
@encryptblockr I also struggle with understanding your request. Maybe you could give an example to make things more clear.

**Comment by oderwat** [comment]
@encryptblockr Did you maybe miss that using the `%%` syntax is optional?

<img width="591" alt="image" src="https://github.com/janpfeifer/gonb/assets/719156/b5bcf03e-c9ee-4955-91ac-3aedbb031090">

and

<img width="523" alt="image" src="https://github.com/janpfeifer/gonb/assets/719156/1b7fa676-9ffa-430e-82c5-de9d8666ec78">

Are just working fine and are a what goes into a `main.go` file.
